### PR TITLE
Qemu guest tools for windows

### PR DESCRIPTION
**Pull request by jonasfj**
@petemoore, if you get bored and have the energy it would be great to get the things you learned from generic-worker on windows into the `system` package... specifically running under different user accounts, creating accounts, etc...

Note: I'll fix the test as best possible, I think it's just the govendor thing...

### CODE REVIEW

**Comment by jonasfj** [comment]
We'll fix OS X tests some other day, I think it's the TTY that doesn't work... I've just skipped the ones I added that didn't work...

All existing tests pass...

File: engines/native/system/process_windows.go [hunk 1]
```diff
@@ -1,40 +1,167 @@
...
+		if err != nil {
+			return nil, err
+		}
+
+		// If the passed owner matches the current user, we set owner to
+		// nil to allow non-root accounts to succeed.
+		if currentUser.Name() == options.Owner.Name() {
+			options.Owner = nil
+		}
+	}
```
Comment (walac) [inline engines/native/system/process_windows.go]: I think the issue doesn't apply to Windows, but not harmful keeping it.

File: engines/native/system/process_windows.go [hunk 2]
```diff
@@ -1,40 +1,167 @@
...
 // KillProcessTree will kill root and all of its descendents.
 func KillProcessTree(root *Process) error {
-	panic("Not implemented")
+	// See:
+	// https://technet.microsoft.com/en-us/library/bb491009.aspx
+	// https://ss64.com/nt/taskkill.html
+	err := exec.Command(
+		`c:\Windows\system32\taskkill.exe`,
+		"/F", "/T", "/PID", strconv.Itoa(root.cmd.Process.Pid),
+	).Run()
```
Comment (walac) [inline engines/native/system/process_windows.go]: Doesn't [`Process.Kill`](https://golang.org/pkg/os/#Process.Kill) work?

File: engines/native/system/system_safe_test.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+	"github.com/taskcluster/taskcluster-worker/runtime/ioext"
+)
+
+// This file contains test cases that don't leave risk leaving dirty system
```
Comment (walac) [inline engines/native/system/system_safe_test.go]: s/leave//

File: engines/native/system/system_safe_test.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+// resources, such as temporary user accounts. However, this file doesn't
+// test all the functionality in the system package. Useful, we you want to
```
Comment (walac) [inline engines/native/system/system_safe_test.go]: "Usually, we want to"?

File: engines/native/system/system_safe_test.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+		p, err := StartProcess(ProcessOptions{
+			Arguments: testFalse,
+			TTY:       true,
+		})
+		require.NoError(t, err)
+		require.False(t, p.Wait())
+	})
+
+	t.Run("StartProcess Cat", func(t *testing.T) {
+		t.Skip("Not implemented on windows, we're missing a testCat command that works")
```
Comment (walac) [inline engines/native/system/system_safe_test.go]: But we are not skipping only on Windows, are we?

File: engines/native/system/system_safe_test.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+			Stdin:     ioutil.NopCloser(bytes.NewBufferString("hello-world")),
+			Stdout:    ioext.WriteNopCloser(&out),
+		})
+		require.NoError(t, err)
+		require.True(t, p.Wait())
+		require.EqualValues(t, "hello-world", out.String())
+	})
+
+	t.Run("StartProcess Cat with TTY", func(t *testing.T) {
+		t.Skip("Not implemented on windows, we're missing a testCat command that works")
```
Comment (walac) [inline engines/native/system/system_safe_test.go]: Same here

File: engines/native/system/system_safe_test.go [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+			WorkingFolder: homeDir,
+		})
+		require.NoError(t, err)
+		require.True(t, p.Wait())
+		require.Contains(t, out.String(), homeDir)
+	})
+
+	t.Run("StartProcess TTY Print Dir", func(t *testing.T) {
+		if rt.GOOS == "darwin" {
+			t.Skip("TODO: fix test can on OS X, no idea why it fails")
```
Comment (walac) [inline engines/native/system/system_safe_test.go]: s/can//

File: engines/native/system/system_safe_test.go [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+			Owner:     u,
+		})
+		require.NoError(t, err)
+		require.True(t, p.Wait())
+		require.Contains(t, out.String(), u.Home())
+	})
+
+	t.Run("StartProcess TTY, Owner and Print Dir", func(t *testing.T) {
+		if rt.GOOS == "darwin" {
+			t.Skip("TODO: fix test can on OS X, no idea why it fails")
```
Comment (walac) [inline engines/native/system/system_safe_test.go]: s/can//

File: engines/native/system/process_windows.go (hunk 2 above)
Comment (jonasfj) [inline engines/native/system/process_windows.go]: it won't kill a process tree

**Pull request merged**